RID.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace dnvm;
+
+internal enum OS
+{
+    win,
+    linux,
+    linux_musl,
+    osx
+}
+
+internal enum Arch
+{
+    x64
+}
+
+internal record struct RID(OS OS, Arch Arch) : IFormattable
```

Comment:
I like the idea of having a RID representation (although by C# standard style it should be called `Rid`. I also wouldn't bother with `IFormattable` because we don't use the format provider for anything. Overriding ToString is enough.